tmt/policy.py

```diff
@@ -0,0 +1,222 @@
+            key: str,
+            template: str,
+            current_value_exported: Any,
+            current_value_source: FieldValueSource,
+            normalize_callback: tmt.container.NormalizeCallback[T],
+        ) -> T:
+            """
+            Update a single key of the object.
+
+            :param key: name of the key to update.
+            :param template: template rendering to a new value.
+            :param current_value_exported: current value as if exported
+                via :py:meth:`Core._export`. Consists of Python built-in
+                types only, no custom classes.
+            :param current_value_source: origin of the current value.
+            :param normalize_callback: will be called with the rendered
+                ``template`` to produce new value for the key.
+            """
+
+            rendered_new_value = render_template(
+                template,
+                VALUE=current_value_exported,
+                VALUE_SOURCE=current_value_source,
+            )
+
+            raw_new_value = tmt.utils.yaml_to_python(rendered_new_value)
+
+            new_value = normalize_callback('', raw_new_value, logger)
+
+            setattr(obj, tmt.container.option_to_key(key), new_value)
+
+            return new_value
+
+        for key, template in self.__dict__.items():
+            logger = base_logger.clone()
+
+            _, _, _, _, field_metadata = tmt.container.container_field(obj, key)
+
+            normalize_callback: Optional[tmt.container.NormalizeCallback[Any]] = (
+                field_metadata.normalize_callback
+            )
+            export_callback: Optional[tmt.container.FieldExporter[Any]] = (
+                field_metadata.export_callback
+            )
+
+            if normalize_callback is None:
+                logger.warning(f"key '{key}' lacks normalizer")
+
+                normalize_callback = lambda key_address, value, logger: value  # noqa: E731
+
+            if export_callback is None:
+                logger.warning(f"key '{key}' lacks exporter")
+
+                export_callback = lambda value: value  # noqa: E731
+
+            current_value = old_value = getattr(obj, tmt.container.option_to_key(key))
+            current_value_exported = old_value_exported = export_callback(current_value)
+            current_value_source = old_value_source = obj._field_value_sources[key]
+
+            if isinstance(
+                current_value,
+                (float, int, bool, str, list, dict, ShellScript, tmt.utils.Environment),
+            ):
+                current_value = set_key(
+                    key,
+                    template,
+                    current_value_exported,
+                    current_value_source,
+                    normalize_callback,
+                )
+
+            else:
+                raise tmt.utils.GeneralError(
+                    f"Field '{key}' of type '{type(current_value)}' is not supported by a policy."
```

Comment:
```suggestion
                    f"Field '{key}' of type '{type(current_value)}' is not supported by the policy."
```

Plus maybe include the `policy` path in the exception?